modules/library/metadata/src/main/java/org/geotools/util/factory/Hints.java

```diff
@@ -1169,12 +1169,12 @@ public static void scanSystemProperties() {
      *     otherwise.
      */
     private static boolean ensureSystemDefaultLoaded() {
-        if (needScan.get()) {
+        // update GLOBAL atomically only if needScan == true
+        if (needScan.compareAndSet(true, false)) {
```

Comment:
Won't this cause a race condition though? Idea:
- Thread1 gets in, needScan is true, sets it to false before doing the scan
- Thread2 gets in and thinks no scan is needed, and will proceeed doing whatever is doing without the system hints being actually loaded yet.

Reply:
You're right. I'm not completely certain this needs to be fixed right now though, will need to re-run the profiling sessions I did before to create the other tickets under GEOT-6420 to see if the extra complexity would be justified. By that I mean that concurrent calls to `Hints.ensureSystemDefaultLoaded()` ultimately get blocked by the `synchronized` block at `GeoTools.scanForSystemHints()` and could be alleviated if need be, but I'll wait till I can justify it or close this instead.